examples/notebooks/persistent_dataset_speed.ipynb

```diff
@@ -0,0 +1,463 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_transforms = set_deterministic(train_transforms)\n",
+    "cache_init_start = time.time()\n",
+    "cache_train_ds = CacheDataset(data=train_files, transform=train_transforms, cache_rate=1.0)\n",
+    "cache_val_ds = CacheDataset(data=val_files, transform=val_transforms, cache_rate=1.0)\n",
+    "cache_init_time = time.time() - cache_init_start\n",
+    "\n",
+    "cache_epoch_num, cache_total_time, cache_epoch_loss_values, cache_metric_values, cache_epoch_times = \\\n",
+    "    train_process(cache_train_ds, cache_val_ds)\n",
+    "print('Total training time of {} epochs with CacheDataset: {:.4f}'.format(cache_epoch_num, cache_total_time))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Plot training loss and validation metrics"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure('train', (12, 12))\n",
+    "plt.subplot(3, 2, 1)\n",
+    "plt.title('Regular Epoch Average Loss')\n",
+    "x = [i + 1 for i in range(len(epoch_loss_values))]\n",
+    "y = epoch_loss_values\n",
+    "plt.xlabel('epoch')\n",
+    "plt.grid(alpha=0.4, linestyle=':')\n",
+    "plt.plot(x, y, color='red')\n",
+    "\n",
+    "plt.subplot(3, 2, 2)\n",
+    "plt.title('Regular Val Mean Dice')\n",
+    "x = [i + 1 for i in range(len(metric_values))]\n",
+    "y = cache_metric_values\n",
+    "plt.xlabel('epoch')\n",
+    "plt.grid(alpha=0.4, linestyle=':')\n",
+    "plt.plot(x, y, color='red')\n",
+    "\n",
+    "plt.subplot(3, 2, 3)\n",
+    "plt.title('PersistentDataset Epoch Average Loss')\n",
+    "x = [i + 1 for i in range(len(epoch_loss_values))]\n",
+    "y = persistence_epoch_loss_values\n",
+    "plt.xlabel('epoch')\n",
+    "plt.grid(alpha=0.4, linestyle=':')\n",
+    "plt.plot(x, y, color='blue')\n",
+    "\n",
+    "plt.subplot(3, 2, 4)\n",
+    "plt.title('PersistentDataset Val Mean Dice')\n",
+    "x = [i + 1 for i in range(len(metric_values))]\n",
+    "y = persistence_metric_values\n",
+    "plt.xlabel('epoch')\n",
+    "plt.grid(alpha=0.4, linestyle=':')\n",
+    "plt.plot(x, y, color='blue')\n",
+    "\n",
+    "plt.subplot(3, 2, 5)\n",
+    "plt.title('Cache Epoch Average Loss')\n",
+    "x = [i + 1 for i in range(len(epoch_loss_values))]\n",
+    "y = cache_epoch_loss_values\n",
+    "plt.xlabel('epoch')\n",
+    "plt.grid(alpha=0.4, linestyle=':')\n",
+    "plt.plot(x, y, color='green')\n",
+    "\n",
+    "plt.subplot(3, 2, 6)\n",
+    "plt.title('Cache Val Mean Dice')\n",
+    "x = [i + 1 for i in range(len(metric_values))]\n",
+    "y = cache_metric_values\n",
+    "plt.xlabel('epoch')\n",
+    "plt.grid(alpha=0.4, linestyle=':')\n",
+    "plt.plot(x, y, color='green')\n",
+    "\n",
+    "plt.show()"
```

Comment:
A suggestion: could you please help also attach the plot charts?
If you don't have available devices to train it now, I can help do it when merged.
Thanks.